src/Supervisor.php

```diff
@@ -225,10 +225,6 @@ public function terminate($status = 0)
             });
         });
 
-        while ($this->processPools->map->runningProcesses()->collapse()->count()) {
```

Comment:
No need to wait for children any more. They have already received the SIGTERM signal, and we made sure no other mechanism causes them to terminate while handling the current job.